File: 프로그래머스/2/42842. 카펫/카펫.java
class Solution {
    public int[] solution(int brown, int yellow) {
        for (int width = 3; width <= 5000; width++) {
            for (int height = 3; height <= width; height++) {
                int boundary = (width + height - 2) * 2;
                int center = (width * height) - boundary;
                
                if (boundary == brown && center == yellow) {
                    return new int[] {width, height};
                }
            }
        }
        return new int[]{-1, -1};
    }
}

File: 프로그래머스/2/42842. 카펫/카펫.cpp
#include <string>
#include <vector>
#include <algorithm>
using namespace std;

vector<int> solution(int brown, int yellow) {
    vector<int> answer;
    
    int total = brown + yellow;
    
    for(int row = 3; row <= total/3; ++row) {
        int col = total / row;
        if (col * row == total && (row - 2) * (col - 2) == yellow) {
            answer.push_back(row);
            answer.push_back(col);
            break;
        }
    }
    
    sort(answer.rbegin(), answer.rend());
    
    return answer;
}